qa-tests-backend/src/test/groovy/VulnMgmtSACTest.groovy

```diff
@@ -100,25 +251,57 @@ class VulnMgmtSACTest extends BaseSpecification {
         return token.token
     }
 
+    def getImageCVEQuery() {
+        return isPostgresRun() ? GET_IMAGE_CVES_QUERY : GET_CVES_QUERY
+    }
+
+    def getNodeCVEQuery() {
+        return isPostgresRun() ? GET_NODE_CVES_QUERY : GET_CVES_QUERY
+    }
+
+    def getImageComponentQuery() {
+        return isPostgresRun() ? GET_IMAGE_COMPONENTS_QUERY : GET_COMPONENTS_QUERY
+    }
+
+    def getNodeComponentQuery() {
+        return isPostgresRun() ? GET_NODE_COMPONENTS_QUERY : GET_COMPONENTS_QUERY
+    }
+
     @Retry(count = 0)
     @Unroll
-    @IgnoreIf({ Env.CI_JOBNAME.contains("postgres") })
-    def "Verify role based scoping on vuln mgmt: #roleName #baseQuery"() {
+    def "Verify role based scoping on vuln mgmt: node-role Node:*"() {
         when:
         "Get Node CVEs and components"
         BaseService.useBasicAuth()
+        def baseQuery = "Node:*"
+        def cveQuery = getNodeCVEQuery()
+        def componentQuery = getNodeComponentQuery()
         def gqlService = new GraphQLService()
-        def baseVulnCallResult = gqlService.Call(GET_CVES_QUERY, [query: baseQuery])
+        def baseVulnCallResult = gqlService.Call(cveQuery, [query: baseQuery])
         assert baseVulnCallResult.hasNoErrors()
-        def baseComponentCallResult = gqlService.Call(GET_COMPONENTS_QUERY, [query: baseQuery])
+        def baseComponentCallResult = gqlService.Call(componentQuery, [query: baseQuery])
         assert baseComponentCallResult.hasNoErrors()
 
         and:
-        gqlService = new GraphQLService(getToken(roleName, roleName))
-        def vulnCallResult = gqlService.Call(GET_CVES_QUERY, [query: ""])
+        gqlService = new GraphQLService(getToken(NODE_ROLE, NODE_ROLE))
+        def vulnCallResult = gqlService.Call(cveQuery, [query: ""])
         assert vulnCallResult.hasNoErrors()
-        def componentCallResult = gqlService.Call(GET_COMPONENTS_QUERY, [query: ""])
+        def componentCallResult = gqlService.Call(componentQuery, [query: ""])
         assert componentCallResult.hasNoErrors()
+        def baseSortedVulns = extractCVEsAndSort(baseVulnCallResult.value)
+        def sortedVulns = extractCVEsAndSort(vulnCallResult.value)
+        if ( baseSortedVulns != sortedVulns ) {
+            for ( v in baseSortedVulns ) {
+                if ( ! sortedVulns.contains(v) ) {
+                    log.info("Item found in baseVulnCallResult but not in vulnCallResults: " + v)
+                }
+            }
+            for ( v in sortedVulns ) {
+                if ( ! baseSortedVulns.contains(v) ) {
+                    log.info("Item found in vulnCallResults but not in baseVulnCallResult: " + v)
+                }
+            }
+        }
```

Comment:
How about something like this?
```suggestion
        def baseSortedVulns = extractCVEsAndSort(baseVulnCallResult.value).toSet()
        def sortedVulns = extractCVEsAndSort(vulnCallResult.value).toSet()
        if ( baseSortedVulns != sortedVulns ) {
            log.info("Item found in baseVulnCallResult but not in vulnCallResults: " + baseSortedVulns-sortedVulns)
            log.info("Item found in vulnCallResults but not in baseVulnCallResult: " + sortedVulns-baseSortedVulns)
        }
```
Why do we need to log it? If it's not always happening the it should be warning or error

Reply:
Logging as error sounds good.
At the point where the test runs in e2e, there are about 5k vulnerabilities in the storage. Finding out which one is triggering the diff is not something doable manually anymore.